include/uxr/agent/client/ProxyClient.hpp

```diff
@@ -35,7 +35,8 @@ class ProxyClient : public std::enable_shared_from_this<ProxyClient>
 
     explicit ProxyClient(
             const dds::xrce::CLIENT_Representation& representation,
-            Middleware::Kind middleware_kind = Middleware::Kind(0));
+            Middleware::Kind middleware_kind = Middleware::Kind(0),
+            std::unordered_map<std::string, std::string> properties = {});
```

Comment:
```suggestion
            std::unordered_map<std::string, std::string> properties = {});
```
If you want to default this to empty, you cannot use & reference, I guess that is why you did this, but it is better to use a pointer defaulted to `nullptr`. Passing a std::map by value can result in a very large copy being performed, since we do not know the size of that map